---
title: "Pipeline Metrics Summary"
author: "Mike Connelly"
date: "2023-08-07"
output: html_document
---
## Setup and packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, cache = TRUE)
options(stringsAsFactors = FALSE)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
```
```{r packages}
library("tidyverse")
#
source("./R/pocillopora_hDNA_uces_functions.R")
```

## Import sample metadata
```{r import_samples_metadata}
samples <- read_csv("./data/specimens.csv")
```

## Import/inspect genome skimming pipeline QC metrics
  Note: Run MultiQC on R1 reads only to create reports for historical samples (n = 12/14)
        
### Raw reads
```{r import_multiqc_data}
# raw reads
multiqc_fastqc <- read_tsv("./outputs/QCs/fastqcs/multiqc_data/multiqc_fastqc.txt") %>%
  mutate("Sample ID" = gsub("_S.*_R[12]_001", "", Sample)) %>%
  relocate(`Sample ID`, .before = `Sample`)
```
```{r inspect_read_depth}
# quick check of sample depth
pdf("./outputs/figures/total_raw_reads.pdf", width = 6.5, height = 4.76)
multiqc_fastqc %>%
  mutate(`Sample ID` = fct_reorder(`Sample ID`, desc(`Total Sequences`))) %>%
  ggplot(aes(`Sample ID`, `Total Sequences`)) +
  geom_point(size = 0.5) +
  geom_hline(yintercept = 1000000, color = "red") +
  theme(axis.text.x = element_text(angle = 90)) +
  ggtitle("Total number of Raw Reads per sample")
dev.off()
```
```{r}
mean(multiqc_fastqc$`Total Sequences`)
sd(multiqc_fastqc$`Total Sequences`)
```


### Trimmed reads
```{r}
# trimmed reads
multiqc_trimqc <- read_tsv("./outputs/QCs/trimqcs/multiqc_data/multiqc_fastqc.txt") %>%
  mutate("Sample ID" = gsub("_S.*_R[12]_PE", "", Sample)) %>%
  relocate(`Sample ID`, .before = `Sample`)

```
```{r inspect_read_Depth}
multiqc_trimqc %>%
  mutate(Sample = fct_reorder(Sample, desc(`Total Sequences`))) %>%
  ggplot(aes(Sample, `Total Sequences`)) +
  geom_point(size = 0.5) +
  geom_hline(yintercept = 1000000, color = "red") +
  theme(axis.text.x = element_text(size = 5, angle = 90))
```



## Read QC, trim, coral alignment pipeline metrics plot
### Summarize QC, trim, alignment data
```{r tidy_multiqc_data}
# tidy multiqc data to look at raw versus trimmed reads
multiqc_fastqc_reads <- multiqc_fastqc %>% filter(grepl("R1", Sample)) %>% dplyr::select(`Sample ID`, `Total Sequences`) %>% mutate("Raw Reads" = `Total Sequences` * 2, .keep = "unused") 
#
multiqc_trimqc_reads <- multiqc_trimqc %>% filter(grepl("R1", Sample)) %>% dplyr::select(`Sample ID`, `Total Sequences`) %>% mutate("Trimmed Reads" = `Total Sequences` * 2, .keep = "unused")
#
multiqc_flagstat_reads <- multiqc_flagstat %>% dplyr::select(`Sample ID`, `mapped_passed`) %>% 
  rename("Mapped Reads" = `mapped_passed`)
#
multiqc_data <- left_join(multiqc_fastqc_reads, multiqc_trimqc_reads, by = "Sample ID") %>%
  left_join(., multiqc_flagstat_reads, by = "Sample ID")
```
```{r}
multiqc_data[duplicated(multiqc_data$`Sample ID`),]
```

```{r}
multiqc_sample_data <- multiqc_data %>%
  inner_join(samples, by = "Sample ID")

mqc <- multiqc_sample_data %>% pivot_longer(., `Raw Reads`:`Mapped Reads`, names_to = "pipeline_step", values_to = "read_count", names_transform = list("pipeline_step" = as_factor))
#
  mqc %>% ggplot(aes(pipeline_step, read_count)) +
  geom_path(aes(group = `Sample ID`), show.legend = FALSE) +
  geom_point(size = 2, shape = 21, color = "black", alpha = 0.5, stroke = 0.5, show.legend = FALSE) + 
  facet_grid(.~Species, scales = "free") +
  # scale_y_continuous(limits = c(0, 3e7), breaks = c(0, 1e7, 2e7, 3e7)) +
  # scale_color_manual(values = genocolors) +
  # scale_shape_manual(values = colshapes) + 
  # scale_fill_manual(values = genocolors) +
  theme(axis.text.x = element_text(angle = 315, hjust = 0),
        axis.title.x = element_blank()
        # panel.grid.major = element_line(color = "grey92"),
        ) +
  ylab("read count") +
  xlab("pipeline step") + 
  ggtitle("read counts")
```

```{r multiqc_plot_data}
multiqc_plot_data <- multiqc_data %>%
  inner_join(samples, by = "Sample ID") %>%
  pivot_longer(., `Raw Reads`:`Mapped Reads`, names_to = "pipeline_step", values_to = "read_count", names_transform = list("pipeline_step" = as_factor)) %>% 
  dplyr::group_by(`Sample ID`) %>%
  dplyr::mutate(proportion = read_count / read_count[pipeline_step == "Raw Reads"])
# 
# multiqc_plot_data$pipeline_step <- factor(multiqc_plot_data$pipeline_step, levels = c("Raw Reads", "Trimmed Reads", "Mapped Reads"), ordered = TRUE) # , "Uniquely Mapped Reads", "Input Reads", "Assigned Reads"
```
### Pipeline metric plots for manuscript
```{r read_depth_plot}
# All sequenced samples
FASTQCg <- multiqc_plot_data %>% 
  filter(pipeline_step == "Raw Reads") %>%
  ggplot(aes(`Sample ID`, read_count))
FASTQCg + geom_point() + 
  geom_hline(yintercept = 10000000, color  = "green") +
  geom_hline(yintercept = 8000000) + 
  geom_hline(yintercept = 1000000, color = "red")
```

```{r}
pipeQC_counts <- multiqc_plot_data %>%
  # dplyr::filter(pipeline_step != "Input Reads" & pipeline_step != "Assigned Reads") %>%
  ggplot(aes(pipeline_step, read_count)) +
  geom_path(aes(group = `Sample ID`), show.legend = FALSE) +
  geom_point(size = 2, shape = 21, color = "black", alpha = 0.5, stroke = 0.5, show.legend = FALSE) + 
  # facet_grid(.~Species, scales = "free") +
  # scale_y_continuous(limits = c(0, 3e7), breaks = c(0, 1e7, 2e7, 3e7)) +
  # scale_color_manual(values = genocolors) +
  # scale_shape_manual(values = colshapes) + 
  # scale_fill_manual(values = genocolors) +
  theme(axis.text.x = element_text(angle = 315, hjust = 0),
        axis.title.x = element_blank()
        # panel.grid.major = element_line(color = "grey92"),
        ) +
  ylab("read count") +
  xlab("pipeline step") + 
  ggtitle("read counts")
pipeQC_counts
```
```{r}
pdf("./outputs/figures/nobones/PROPread.pdf", width = 6.5, height = 4.76)
pipeQC_proportions <- multiqc_plot_data %>%
  # dplyr::filter(pipeline_step != "Input Reads" & pipeline_step != "Assigned Reads") %>%
  ggplot(aes(pipeline_step, proportion)) +
  geom_path(aes(group = `Sample ID`), show.legend = FALSE) +
  geom_point(size = 2, shape = 21, color = "black", alpha = 0.5, stroke = 0.5, show.legend = FALSE) + 
  # facet_grid(.~Region, scales = "free") +
    # facet_grid(.~Species, scales = "free") +
  scale_y_continuous(limits = c(0.0, 1.0)) +
  # scale_color_manual(values = genocolors) +
  # scale_shape_manual(values = colshapes) + 
  # scale_fill_manual(values = genocolors) +
  theme(axis.text.x = element_text(angle = 315, hjust = 0),
        axis.title.x = element_blank()
        # panel.grid.major = element_line(color = "grey92"),
        ) +
  ylab("proportion raw reads") +
  xlab("pipeline step") + 
  ggtitle("read proportions")
pipeQC_proportions
dev.off()
```



### SPAdes assembly metrics
```{r}

```

### PHYLUCE alignment matrix stats
```{r}

```


SNP matrix stats