---
title: "PHYLUCE_phylogenetic_analysis"
author: "Mike Connelly"
date: "2023-05-24"
output: html_document
---
## Setup and packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, cache = TRUE)
options(stringsAsFactors = FALSE)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
```
```{r packages}
library("tidyverse")
library("treeio")
library("ggtree")
library("phytools")
# 
source("./R/pocillopora_hDNA_uces_functions.R")
```

## Import sample metadata
```{r import_sample_metadata}
samples <- read_csv("./data/specimens.csv")
```
```{r outgroup_metadata}
# outgroups <- read_csv("./data/outgroups.csv")
# samples_out <- rbind(samples, outgroups)
```

## Phylogenetic trees from PHYLUCE UCE alignments

### IQ-TREE - 75% sample occupancy matrix
```{r}
# read-in tree file
tree <- read.newick("./outputs/iqtree/iqtree_contigs300_75.treefile")
# midpoint root
tree <- midpoint.root(tree)
# set root-edge
tree$root.edge <- 1

# adjust tip labels
# remove contigs suffix 
tips <- gsub("_contigs$", "", tree$tip.label)
# remove contigs suffix 
tips <- gsub("_S[[:digit:]]*$", "", tips)
# add underscore to USNM specimens
tips <- gsub("USNM", "USNM_", tips)
# check tips
tips
# assign new tip labels to tree
tree$tip.label <- tips


# basic viz
gg_tree(tree)
```

```{r}
# create dataframe holding metadata 
tips_metadata <- data.frame(tips)
tips_metadata$class <- ifelse(grepl("USNM", tips), "USNM", ifelse(grepl("pocillopora|stylophora", tips), "Genome", "Ref"))
# may want way to identify genome sequences - blue color?
```
```{r}
# node labels with bootstrap support
node.labels <- tree$node.label
node.labels[node.labels == "Root"] <- ""
node.labels[node.labels == "100/100"] <- ""
# define high-support bootstrap values - 90%?
node.labels
#
tree$node.label <- node.labels
```
```{r}
# highlight clades to make it pretty
tree$Nnode
tree$edge.length
tree$root.edge
tree$node.label
```



```{r basic_ggtree_viz}
pdf("./outputs/figures/iqtree_75pct_tree.pdf", width = 8.5, height = 6.5)
#
tree_title <- expression(paste("IQ-TREE Maximum Likelihood Tree with NMNH IZ ", italic("Pocillopora"), " specimens"))
loci_info <- c("75% sample occupancy, 225 UCE loci, 105,205 bp, 6,736 informative sites")
#
tree %>% ggtree(., right = TRUE) %<+% tips_metadata + 
  geom_tiplab(hjust = -0.08, aes(color = `class`, size = `class`)) +
  geom_nodelab(size = 2) +
# figure out way to highlight clades
  geom_rootedge(rootedge = 0.01) +
  geom_treescale(width = 0.02, x = 0, y = 32, linesize = 1.5) +
  ggplot2::xlim(-0.015, 0.35) +
  scale_color_manual(values = c("navyblue", "black", "red")) +
  scale_size_manual(values = c(2.75, 2.75, 3.25)) +
   theme(plot.margin = margin(5,5,2,5, unit = "mm"),
         legend.position = "none") +
  ggtitle(tree_title, subtitle = loci_info)
#
dev.off()
```
```{r}
pdf("./outputs/figures/iqtree_nodes.pdf", width = 11.5, height = 8)
tree %>% ggtree() + geom_text(aes(label=node), size = 2, hjust=.45)
dev.off()
```

```{r annotated_ggtree_viz}
# clades - node 103: P. effusa, node 108: P. grandis, node 180: P. meandrina, node 183: P. verrucosa-like
pdf("./outputs/figures/iqtree_02.pdf", width = 8.5, height = 11.5)
#
tree %>% ggtree(., right = TRUE) %<+% samples + 
  geom_tippoint(aes(fill = `mtorf_type`), shape = 21, alpha = 0.8, stroke = 0.5, color = "black", hjust = 0.3, show.legend = TRUE) +
  geom_tiplab(size = 3, hjust = -0.05) +
  # add bootstrap support values
  geom_nodelab(size = 2, hjust = 1.25, vjust = -0.25) +
  # add clade annotations
  geom_cladelab(node = 103, label = "P. effusa (Clipperton)", angle = 0, 
                  fontsize = 3, offset = 0.15) +
  geom_cladelab(node = 108, label = "P. grandis-like", angle = 0, 
                  fontsize = 3, offset = 0.15) +
  geom_cladelab(node = 180, label = "P. meandrina-like", angle = 0, 
                  fontsize = 3, offset = 0.15) +
  geom_cladelab(node = 183, label = "P. verrucosa-like", angle = 0, 
                  fontsize = 3, offset = 0.15) +
  ggplot2::xlim(-0.05, 0.5) +
  geom_treescale(width = 0.05, x = 0) +
  scale_fill_manual(values = mtorf_colors, name = "mtORF Type") +
   theme(legend.position = "right",
        legend.margin = margin(0,0,0,0, unit = "mm"),
        plot.margin = margin(5,5,5,5, unit = "mm")) +
  ggtitle(tree_title)
dev.off()
```

### IQ-TREE - 50% sample occupancy matrix
```{r}
# read-in tree file
tree <- read.newick("./outputs/iqtree/iqtree_contigs300_50.treefile")
# midpoint root
tree <- midpoint.root(tree)
# set root-edge
tree$root.edge <- 1

# adjust tip labels
# remove contigs suffix 
tips <- gsub("_contigs$", "", tree$tip.label)
# remove contigs suffix 
tips <- gsub("_S[[:digit:]]*$", "", tips)
# add underscore to USNM specimens
tips <- gsub("USNM", "USNM_", tips)
# check tips
tips
# assign new tip labels to tree
tree$tip.label <- tips


# basic viz
gg_tree(tree)
```

```{r basic_ggtree_viz}
pdf("./outputs/figures/iqtree_50pct_tree.pdf", width = 6.5, height = 6.5)
#
tree_title <- expression(paste("IQ-TREE Maximum Likelihood Tree with NMNH IZ ", italic("Pocillopora"), " specimens"))
loci_info <- c("50% sample occupancy, 1,422 UCE loci, 576,543 bp, 30,967 informative sites")
#
tree %>% ggtree(., right = TRUE) %<+% samples + 
  # geom_tippoint(aes(fill = `Species`), shape = 21, alpha = 0.8, stroke = 0.5, color = "black", hjust = 0.3, show.legend = TRUE) +
  geom_tiplab(size = 2.75, hjust = -0.08) +
  geom_rootedge(rootedge = 0.01) +
  geom_treescale(width = 0.05, x = 0, y = 1) +
  ggplot2::xlim(-0.015, 0.35) +
  # scale_fill_manual(values = spp_colors, name = "Species") +
   theme(legend.position = c(0.12, 0.88),
        legend.margin = margin(0,0,0,0, unit = "mm"),
        plot.margin = margin(5,5,5,5, unit = "mm")) +
  ggtitle(tree_title, subtitle = loci_info)
#
dev.off()
```
